### PR TITLE
[15.0][ADD] hr_holidays_leave_self_approve

### PR DESCRIPTION
**Pull request by florentx**

### CODE REVIEW

**Comment by rafaelbn** [comment]
I think an better explanation should be in README as this can be done in Odoo Community without any module by:

- Setting approval to "no validation" or (better)
- Just setting "By Employee's Approver" and changing the domain of the field in the employee to it self.

Just curious, I would not use this module even when I allow my employees self-allow approvals 😄 

Not blocking anything! Regards

**Comment by florentx** [comment]
> I think an better explanation should be in README as this can be done in Odoo Community without any module by:
> 
>     * Setting approval to "no validation" or (better)
> 
>     * Just setting "By Employee's Approver" and changing the domain of the field in the employee to it self.
> 

This add-on is useful for our use case because:

1. They want a validation, even if it is a self validation
2. Changing the domain of the field to the employee itself does not work because of this check in standard (Odoo 15)
```
# File "addons/hr_holidays/models/hr_leave.py"

                    if holiday.employee_id == current_employee:
                        raise UserError(_('Only a Time Off Manager can approve/refuse its own requests.'))
```

File: hr_holidays_leave_self_approve/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Self Approve Leaves",
+    "version": "15.0.1.0.0",
+    "license": "AGPL-3",
+    "summary": "Allow employee to approve his own leaves",
+    "author": "Camptocamp SA, Odoo Community Association (OCA)",
```
Comment (simahawk) [inline hr_holidays_leave_self_approve/__manifest__.py]:
```suggestion
    "author": "Camptocamp, Odoo Community Association (OCA)",
```

File: hr_holidays_leave_self_approve/tests/test_hr_holidays_leave_self_approve.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+
+from odoo.tests.common import TransactionCase
+
+
+class TestHolidaysAutoValidate(TransactionCase):
+    def setUp(self):
```
Comment (simahawk) [inline hr_holidays_leave_self_approve/tests/test_hr_holidays_leave_self_approve.py]: pls use setUpClass and disable tracking

File: hr_holidays_leave_self_approve/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Self Approve Leaves",
+    "version": "15.0.1.0.0",
+    "license": "AGPL-3",
+    "summary": "Allow employee to approve his own leaves",
```
Comment (simahawk) [inline hr_holidays_leave_self_approve/__manifest__.py]:
```suggestion
    "summary": "Allow employees to approve their own leaves when allowed",
```

**Review by simahawk — CHANGES_REQUESTED** [review]

**Review by simahawk — APPROVED** [review]

**Review by gurneyalex — APPROVED** [review]

**Pull request merged**